src/Hypergrid.js

```diff
@@ -263,6 +266,82 @@ Hypergrid.prototype = {
         this.refreshProperties();
     },
 
+    /** @typedef pluginSpec
+     * @type {object|function|Array}
+     * One of:
+     * * simple API - a plain object with an `install` method
+     * * object API - an object constructor
+     * * array:
+     *    * first element is an optional name for the API or the newly instantiated object
+     *    * next element (or first element when not a string) is the simple or object API
+     *    * remaining arguments are optional arguments for the object constructor
+     * * falsy value such as `undefined` - ignored
+     *
+     * The API may have a `name` or `$$CLSSS_NAME` property.
+     */
+    /**
+     * @desc Plugin installation:
+     * * Each simple API is installed by calling it's `install` method with `this` as first arg + any additional args listed in the `pluginSpec` (when it is an array).
+     * * Each object API is installed by instantiating it's constructor with `this` as first arg + any additional args listed in the `pluginSpec` (when it is an array).
+     *
+     * The resulting plain object or instantiated objects may be named by (in priority order):
+     * 1. if `pluginSpec` contains an array and first element is a string
+     * 2. object has a `name` property
+     * 3. object has a `$$CLASS_NAME` property
```

Comment:
Should consider if extend-me specific properties will cause issues if we ever switch to ES6